vespalib/src/vespa/vespalib/datastore/fixed_size_hash_map.h

```diff
@@ -0,0 +1,105 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "atomic_entry_ref.h"
+#include <vespa/vespalib/util/array.h>
+#include <vespa/vespalib/util/arrayref.h>
+#include <vespa/vespalib/util/generationhandler.h>
+#include <limits>
+#include <atomic>
+#include <deque>
+#include <functional>
+
+namespace vespalib { class GenerationHolder; }
+namespace vespalib::datastore {
+
+class EntryComparator;
+
+/*
+ * Fixed sized hash map over keys in data store.
```

Comment:
Please add more details on the implementation, how it relates to a data store, what is the key and value, and how to use it (e.g. how it relates to writer / readers and generation handling).